frontend/src/core/resource/components/ResourceMenu.js

```diff
@@ -0,0 +1,164 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from 'fluent-react';
+import onClickOutside from 'react-onclickoutside';
+
+import './ResourceMenu.css';
+
+import type { NavigationParams } from 'core/navigation';
+import type { Resource, ResourcesState } from '..';
+
+
+type Props = {|
+    parameters: NavigationParams,
+    resources: ResourcesState,
+    navigateToPath: (string) => void,
+|};
+
+type State = {|
+    visible: boolean,
+|};
+
+
+/**
+ * Render a resource menu for the main navigation bar.
+ *
+ * Allows to switch between resources without reloading the Translate app.
+ */
+export class ResourceMenuBase extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            visible: false,
+        };
+    }
+
+    toggleVisibility = () => {
+        this.setState((state) => {
+            return { visible: !state.visible };
+        });
+    }
+
+    // This method is called by the Higher-Order Component `onClickOutside`
+    // when a user clicks outside the user menu.
+    handleClickOutside = () => {
+        this.setState({
+            visible: false,
+        });
+    }
+
+    navigateToPath = (event: SyntheticMouseEvent<>) => {
+        event.preventDefault();
+
+        // $FLOW_IGNORE
+        const path = event.currentTarget.pathname;
+        this.props.navigateToPath(path);
+
+        this.setState({
+            visible: false,
+        });
+    }
+
+    renderPercent = (resource: Resource) => {
+        return Math.floor(resource.approved_strings / resource.total_strings * 100) + '%';
+    }
+
+    renderResource = (resource: Resource, i: number) => {
```

Comment:
I've actually come to think that if you create a `renderX` method, you actually want to have an `X` component. In this case, it would be perfect to use a simple function component (instead of a class), because the code is so small. I'm more and more thinking that we should drop the rule of using only class-based components and embrace function components and hooks. What do you think?

Reply:
Yeah I was thinking about a component and I thought it was an overkill, but that's before I've been introduced to the concept of functional components. :) I'll give it a try.